src/controllers/urls.ts

```diff
@@ -28,13 +28,59 @@ export const createUrl = async (
       }
       const groupCode = splitShortCode[0]
 
-      const group = await Groups.findCreateFind({
-        where: { prefix: groupCode },
+      var group: GroupAttributes | null
+      var groupCreated: boolean
+      if (user.role === 'intern') {
+        group = await Groups.find({
+          where: {
+            prefix: groupCode
+          }
+        })
+      } else {
+        [group, groupCreated] = await Groups.findCreateFind({
+          where: { prefix: groupCode },
+          defaults: {
+            prefix: groupCode,
+            ownerId: user.id,
+          },
+        })
+      }
+      if (!group) {
+        throw new Error('Could not find Group Code')
+      }
+
+      const opts = optsFromGroupedShortcode(group, splitShortCode[1])
+      const [url, urlCreated] = await URLs.findCreateFind({
```

Comment:
`URLs.findCreateFind` is repeated everywhere. We do not need that. We just want to have `opts` separately created, the code after that is common